profiles/tests/test_gql_federation.py

```diff
@@ -3,54 +3,71 @@
 from guardian.shortcuts import assign_perm
 
 from open_city_profile.tests.asserts import assert_match_error_code
-from profiles.schema import ProfileNode
-from profiles.tests.factories import ProfileFactory
+from profiles.schema import AddressNode, ProfileNode
+from profiles.tests.factories import AddressFactory, ProfileFactory
 from services.tests.factories import ServiceConnectionFactory
 
+GRAPHQL_SDL_QUERY = """
+    query {
+        _service {
+            sdl
+        }
+    }
+"""
 
-def test_profile_node_exposes_key_for_federation_gateway(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+
+PROFILE_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on ProfileNode {
+                id
+                firstName
             }
         }
-    """
-
-    executed = anon_user_gql_client.execute(query)
-    assert (
-        'type ProfileNode implements Node  @key(fields: "id")'
-        in executed["data"]["_service"]["sdl"]
-    )
+    }
+"""
 
 
-def test_profile_connection_schema_matches_federated_schema(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+ADDRESS_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on AddressNode {
+                id
+                address
+                postalCode
             }
         }
-    """
+    }
+"""
 
-    executed = anon_user_gql_client.execute(query)
+
+@pytest.mark.parametrize(
+    "schema_type, is_extendable", [("ProfileNode", True), ("AddressNode", True)],
```

Comment:
The `is_extendable` parameter doesn't seem to have any use. Perhaps it can be removed.

Reply:
Actually `is_extendable` can be used to also test for nodes that are not entities. I've added 2 sample nodes that aren't extendable in 467f9aa to illustrate the usecase. However, I can remove it if you see fit 👍🏼

Reply:
I prefer just removing `is_extendable`.

Reply:
Sure. Done in 4683281a67ebe191c7a540b3ba5178d372149de9 👍🏼

Reply:
Nice. Could you squash 4683281a67 into 05996262d6 and then I think this would be done.

Reply:
Ok, I have force-pushed all commits from fe68ac7c09485bf29e4e34286cf1edff4660c584 onwards 👍🏼